src/viadot/orchestration/prefect/flows/transform_and_catalog.py

```diff
@@ -273,4 +277,14 @@ def transform_and_catalog(  # noqa: PLR0913, PLR0915
     )
     remove_dbt_repo_dir(dbt_repo_name, wait_for=wait_for)
 
-    return Failed() if task_failed else remove_dbt_repo_dir
+    if task_failed:
+        return Failed()
+
+    if fail_flow_on_test_failure:
```

Comment:
Shouldn't you have added 
```suggestion
    if not fail_flow_on_test_failure:
```
?

Reply:
Good catch, thanks! Fixed in https://github.com/dyvenia/viadot/pull/1173/commits/131db256d3f46d3bc74ab607e2e4a99361a53e02